crates/router/src/connector/adyen.rs

```diff
@@ -733,9 +733,13 @@ impl api::IncomingWebhook for Adyen {
         let notif = get_webhook_object_from_body(request.body)
             .change_context(errors::ConnectorError::WebhookEventTypeNotFound)?;
 
-        Ok(match notif.event_code.as_str() {
-            "AUTHORISATION" => api::IncomingWebhookEvent::PaymentIntentSuccess,
-            _ => Err(errors::ConnectorError::WebhookEventTypeNotFound).into_report()?,
+        Ok(match notif.event_code {
+            adyen::WebhookEventCode::Authorisation => {
+                api::IncomingWebhookEvent::PaymentIntentSuccess
+            }
+            adyen::WebhookEventCode::Refund => api::IncomingWebhookEvent::RefundSuccess,
+            adyen::WebhookEventCode::CancelOrRefund => api::IncomingWebhookEvent::RefundSuccess,
+            adyen::WebhookEventCode::RefundFailed => api::IncomingWebhookEvent::RefundFailure,
```

Comment:
This can be converted to `From` implementation?

Reply:
Done